### ISSUE TITLE
Mock object with [String: AnyHashable] data doesn't compile

### ISSUE THREAD

**Issue by Cookiezby** [apollographql/apollo-ios#3264]
### Summary

In the official site, the mock object can be initialized with DataDict type which receive data with type [String: AnyHashable], but it will throw compile error 

`Cannot convert value of type '[String : Any]' to expected dictionary value type 'String' `

### Version

1.6.1

### Steps to reproduce the behavior

It is not related to specific version of Apollo, but maybe related to Swift language, I am testing with Swift 5.9, the snippet from documentation site can not compile correctly

```swift
let data: [String: AnyHashable] = [
    "data": [
        "__typename": "Query",
        "hero": [
            "__typename": "Human",
            "id": "123",
            "name": "Obi-wan Kenobi",
            "friends": [
                [
                    "__typename": "Human",
                    "id": "456",
                    "name": "Padme Amidala"
                ],
                [
                    "__typename": "Droid",
                    "id": "789",
                    "name": "C-3PO"
                ]
            ]
        ]
    ]
]
```
the error says `Cannot convert value of type '[String : Any]' to expected dictionary value type 'String'`

If you change the type from **`[String: AnyHashable]`** to **`[String: Any]`** , it won't throw any issue, but the DataDict can not receive **`[String: Any]`**

**Comment by Cookiezby** [comment]
I found it needs to define every nested dictionary as [String: AnyHashable] so that the data can be [String: AnyHashable]

```swift
let frind: [String: AnyHashable] =  [
    "__typename": "Human",
    "id": "456",
    "name": "Padme Amidala"
]

let hero: [String: AnyHashable] = [
    "__typename": "Human",
    "id": "123",
    "name": "Obi-wan Kenobi",
    "friends": [
        frind
    ]
]

let query: [String: AnyHashable] = [
    "__typename": "Query",
    "hero": hero
]

let data: [String: AnyHashable] = [
    "data": query
]
```

**Comment by bdbergeron** [comment]
This is a Swift language/compiler issue and not inherent to `apollo-ios`. Try doing this instead:

```swift
let data: [String: AnyHashable] = [
  "data": [
    "__typename": "Query",
    "hero": [
      "__typename": "Human",
      "id": "123",
      "name": "Obi-wan Kenobi",
      "friends": [
        [
          "__typename": "Human",
          "id": "456",
          "name": "Padme Amidala"
        ],
        [
          "__typename": "Droid",
          "id": "789",
          "name": "C-3PO"
        ]
      ]
    ] as [String: AnyHashable]
  ] as [String: AnyHashable]
]
```

**Comment by AnthonyMDev** [comment]
We've merged in PR [#102](https://github.com/apollographql/apollo-ios-dev/pull/102) to fix this issue. Will be pushed out in the next release!
